### PR TITLE
Implement lastModifiedDate tracking support

### PR DESCRIPTION
**Pull request by jujaga**
# Description

This change begins to track the LastModified attribute in S3 versions. However, due to the HeadObjectCommand returning a 404/405 when asking about a delete marker, we will not be able to back-populate delete marker version records on the fly. The synchronization operation is not affected by this shortcoming as it uses a different pathway to surface this data.

[SHOWCASE-3505](https://apps.nrs.gov.bc.ca/int/jira/browse/SHOWCASE-3505)

## Types of changes

New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have checked that unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: app/src/services/version.js [hunk 1]
```diff
@@ -90,7 +90,8 @@ const service = {
           createdBy: userId,
           deleteMarker: data.deleteMarker,
```
Comment (TimCsaky) [inline app/src/services/version.js:91]: I think you need to add lastModifiedDate to the db insert happening on line 48 of services/version.js when it does the copy process. This is invoked during the flow when updating/replacing metadata when the bucket doesnt have versioning.

Comment (TimCsaky) [reply app/src/services/version.js:91]: separate note.. i hope the lastModifiedDate for a version doesnt change when you edit tags on that version 😬

File: app/src/controllers/object.js [hunk 2]
```diff
@@ -344,6 +346,11 @@ const controller = {
         }
       }
 
+      const s3Head = await storageService.headObject({
+        filePath: joinPath(bucketKey, req.currentUpload.filename),
+        bucketId: bucketId
+      }).catch(() => ({}));
```
[the remark this answers is not in the document]
Comment (jujaga) [reply app/src/controllers/object.js:352]: The intent here is to silently fail and continue in the event the last modified date is not acquirable. This is an expected scenario in the following situations:
- The latest object is a deleteMarker: in this situation, we can expect a 404 response, which if not caught will blow up the stack. We provide and return a silent empty object in order to ensure that the request proceeds even without the last modified date.
- There is an errant network issue that makes this call throw. As it is not on the critical path to be able to complete the operation at this stage, we do a silent catch and return in order to make sure the request code continues to execute.

File: app/src/controllers/object.js [hunk 3]
```diff
@@ -1086,6 +1097,11 @@ const controller = {
...
```
[the remark this answers is not in the document]
Comment (jujaga) [reply app/src/controllers/object.js:1103]: The intent here is to silently fail and continue in the event the last modified date is not acquirable. This is an expected scenario in the following situations:
- The latest object is a deleteMarker: in this situation, we can expect a 404 response, which if not caught will blow up the stack. We provide and return a silent empty object in order to ensure that the request proceeds even without the last modified date.
- There is an errant network issue that makes this call throw. As it is not on the critical path to be able to complete the operation at this stage, we do a silent catch and return in order to make sure the request code continues to execute.

File: app/src/services/version.js (hunk 1 above)
Comment (jujaga) [reply app/src/services/version.js:91]: Good catch. Looking into adding and updating the copy service to account for the new value. Will investigate tag modification signatures.

**Review by kyle1morel — APPROVED** [review]

**Pull request merged**